plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java

```diff
@@ -580,7 +581,7 @@ private void configureWith(List<YamlSource> sources) throws ConfiguratorExceptio
             return Files.list(root)
                     .filter(Files::isRegularFile) // only consider regular files, following symlinks
                     .filter(matcher::matches)     // matching pattern
-                    .filter(a -> !a.equals("secrets.yml")) // exclude secrets.yml
+                    .filter(f -> !f.equals(YamlSecretSource.DEFAULT_JENKINS_YAML_PATH)) // exclude secrets.yml
```

Comment:
😅 Well what if it is not the Default YamlSecretSource Path 🤭

Reply:
I agree.  I guess the only way to fix it would be to define the YamlSource in that class as well ...  How would you solve it ?

Reply:
I'm wondering if the filter can be ommitted after all.  Will the configuration of jenkins fail if the secrets block would end up merged in the main config file ?

Reply:
You could omit the secrets block from the JCasc instead of trying to do it at the file level?

Reply:
You mean in YamlUtils.merge ?

Reply:
Modified the code to skip the secrets source during merging in YamlUtils